bids/layout/layout.py

```diff
@@ -546,6 +546,38 @@ def _get_fieldmaps(self, path):
                     fieldmap_set.append(cur_fieldmap)
         return fieldmap_set
 
+    def get_tr(self, derivatives=False, **selectors):
+        """ Returns the scanning repetition time (TR) for one or more runs.
+
+        Args:
+            derivatives (bool): If True, also checks derivatives images.
+            selectors: Optional keywords used to constrain the selected runs.
+                Can be any arguments valid for a .get call (e.g., BIDS entities
+                or JSON sidecar keys).
+        
+        Returns: A single float.
+
+        Notes: Raises an exception if more than one unique TR is found.
+        """
+        # Constrain search to functional images
+        selectors['suffix'] = 'bold'
+        selectors['datatype'] = 'func'
+        images = self.get(extensions='.nii.gz', derivatives=derivatives,
+                            **selectors)
+        if not images:
+            raise ValueError("No functional images that match criteria found.")
+        
+        all_trs = []
+        for img in images:
+            md = self.get_metadata(img.path, suffix='bold', full_search=True)
+            all_trs.append(round(float(md['RepetitionTime']), 5))
+ 
+        unique_trs = set(all_trs)
+        if len(unique_trs) > 1:
+            raise ValueError("Unique TR cannot be found given selectors {!r}"
+                             .format(selectors))
+        return list(unique_trs)[0]
```

Comment:
Might want to check for an empty set, as well, but this change will raise exceptions in the same conditions as your existing line:

```suggestion
        return all_trs.pop()
```